Bottom: 2e49cffa99107cc5dddda5b57cf335911350a8cb
Top:    7b6f5422c8cc5ef49ea7778aca6573c64f44be6b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-12 14:43:30 +0100

Add profiling to file load


---

diff --git a/src/file.c b/src/file.c
index 724904c..aad0837 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,10 +350,14 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
 
   /* Hack, load the mapping file */
   RefdesMapInit ();
 
+  start = clock();
+
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
     {
@@ -383,6 +387,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %d\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
